### PR TITLE
Zi/renew ca

### PR DESCRIPTION
**Pull request by lziest**
The aim is to add a method that can takes an expiring root CA certificate and key to generate a new root CA certificate with the same expiry length. The approach is to reuse the initca.NewFromSigner code path as much as possible. So we add a method in package csr to extract CertificateRequest from x509.Certificate.

### CODE REVIEW

File: initca/initca.go [hunk 1]
```diff
@@ -179,6 +227,51 @@ func NewFromSigner(req *csr.CertificateRequest, priv crypto.Signer) (cert, csrPE
...
+// RenewFromSigner re-creates a root certificate from the CA cert and crypto.Signer.
+// The resulting root certificate will have ca
+// as the template and have the same expiry length.
+func RenewFromSigner(ca *x509.Certificate, priv crypto.Signer) ([]byte, error) {
+	if !ca.IsCA {
+		return nil, errors.New("input certificate is not a CA cert")
+	}
+
+	// it is not within renew window of expiration, don't re-generate
+	if time.Now().AddDate(0, 0, RenewWindowInDays).Before(ca.NotAfter) {
```
Comment (grittygrease) [inline initca/initca.go]: Why do we restrict the time period for which we can renew a CA?

**Comment by andrewbuss** [comment]
For later reference, which commands did you use to generate the test data?

**Comment by grittygrease** [comment]
How the expiration time of the new CA is computed should be spelled out somewhere in the docs.

File: initca/initca.go (hunk 1 above)
Comment (lziest) [inline initca/initca.go]: Trying to only have expiring CA cert be re-generated. I am open to remove it. I guess it is better to let caller decide whether to call Renew.

**Comment by lziest** [comment]
@andrewbuss for generating CA certificate, check the new README I added 
https://github.com/cloudflare/cfssl/commit/2ada58a613d4a4f0f36f1d28d074be32b5b45750

File: initca/initca.go (hunk 1 above)
Comment (lziest) [inline initca/initca.go]: Now removed.

**Comment by grittygrease** [comment]
LGTM

**Pull request merged**